src/components/NewRoomModal.tsx

```diff
@@ -0,0 +1,167 @@
+'use client'
+import { useState, useRef, useCallback, useEffect } from 'react'
+import { useRouter } from 'next/navigation'
+
+// import { socket } from '@/socket'
+
+import RightArrowIcon from './icons/RightArrowIcon'
+
+export default function NewRoomModal() {
+	const [isLoading, setIsLoading] = useState(true)
+	const [roomUrl, setRoomUrl] = useState('')
+	const [hostRoomUrl, setHostRoomUrl] = useState('')
+	const [hideGoBtn, setHideGoBtn] = useState(true)
+	const [hideCreateBtn, setHideCreateBtn] = useState(true)
+	const [showNameInput, setShowNameInput] = useState(true)
+	const [domain, setDomain] = useState('')
+	const roomUrlRef = useRef<HTMLSpanElement>(null)
+	const dialogRef = useRef<HTMLDialogElement>(null)
+	const hostRef = useRef<HTMLInputElement>(null)
+	const [hostFirstName, setHostFirstName] = useState('')
+
+	const router = useRouter()
+
+	useEffect(() => {
+		if (typeof window !== 'undefined') {
+			const hostname = window.location.hostname
+			const http = hostname === 'localhost' ? 'http' : 'https'
+			const port = window.location.port ? `:${window.location.port}` : ''
+			setDomain(`${http}://${hostname}${port}`)
+		}
+		const hostDataLocalStorage = localStorage.getItem('scrumDivingHostData')
+		if (hostDataLocalStorage) {
+			const hostData = JSON.parse(hostDataLocalStorage)
+			if (hostData?.hostName) {
+				setHostFirstName(hostData?.hostName)
+				setHideCreateBtn(false)
+				setShowNameInput(false)
+			}
+			console.log('%c>>> hostData LocalStorage', 'color: red', hostData)
+		}
+		setIsLoading(false)
+	}, [])
+
+	function handleOnChange(event: React.ChangeEvent<HTMLInputElement>) {
+		event.preventDefault()
+		setHostFirstName(event.target.value)
+		setHideCreateBtn(false)
+	}
+
+	function generateAlphaNumeric() {
+		return Math.random().toString(36).substring(3)
+	}
+
+	function copyToClipboard() {
+		if (roomUrlRef.current) {
+			navigator.clipboard.writeText(roomUrlRef.current.textContent || '')
+			alert('Room URL copied to clipboard!')
+		}
+		setHideGoBtn(false)
+	}
+
+	const handleShowModal = useCallback(() => {
+		dialogRef.current?.showModal()
+	}, [])
+
+	function handleCreateRoom() {
+		const newRoomNumber = generateAlphaNumeric() + generateAlphaNumeric()
+		setRoomUrl(`${domain}/${newRoomNumber}`)
+		setHostRoomUrl(`${domain}/host/${newRoomNumber}`)
+		if (hostRef.current) hostRef.current.value = ''
+		setHideGoBtn(true)
+		handleShowModal()
+		console.log('%c>>> hostFirstName:', 'color: red', hostFirstName)
+		console.log('%c>>> room:', 'color: #5f0', domain, newRoomNumber)
+	}
+
+	function goToNewRoom() {
+		console.log('%c>>> user room', 'color: red', roomUrl)
+		console.log('%c>>> host room', 'color: yellow', hostRoomUrl)
+		setHideCreateBtn(true)
+		const valuesLocalStorate = JSON.stringify({
```

Comment:
question: Why do you decide to store in localstorage at this moment Don't you think it's better to do that earlier when  the moment the room is created when clicking to create a new room? (before opening the modal)

Reply:
Or even better, the moment the room is created, redirect to that page right away, and then show the link on that same page to share with colleagues at any time. Up to you to decide :)

Reply:
Done

Reply:
This component has been deleted.